### ISSUE TITLE
Recent GooCanvas2::CairoTypes seems to break capture

### ISSUE THREAD

**Issue by kuhnchris** [shutter-project/shutter#347]
#### Brief summary of issue
Updated system, installed latest GooCanvas2 AUR (arch, 2.0.4-9), installed CPAN GooCanvas2::CairoTypes, after that, taking screenshots with "shutter -s" breaks.

#### Steps to reproduce the issue

Unsure.

#### Error output

```
*** unhandled exception in callback:
***   type GooCanvas2::CanvasImage does not support property 'pattern' at /usr/local/share/shutter/resources/modules/Shutter/Screenshot/SelectorAdvanced.pm line 264.
***   	Shutter::Screenshot::SelectorAdvanced::select_advanced(Shutter::Screenshot::SelectorAdvanced=HASH(0x557133a75b30)) called at /usr/local/bin/shutter line 5946
***   	Shutter::App::fct_take_screenshot("global_keybinding", "select", undef, "") called at /usr/local/bin/shutter line 3029
***   	Shutter::App::__ANON__() called at /usr/lib/perl5/5.34/vendor_perl/Glib/Object/Introspection.pm line 67
***   	eval {...} called at /usr/lib/perl5/5.34/vendor_perl/Glib/Object/Introspection.pm line 67
***   	Glib::Object::Introspection::__ANON__(Shutter::App=HASH(0x5571330c1ef8)) called at /usr/local/bin/shutter line 10794
***  ignoring at /usr/lib/perl5/5.34/vendor_perl/Glib/Object/Introspection.pm line 67.
 at /usr/local/bin/shutter line 10794.
```

#### Extra information, such as Shutter version, operating system and ideas for how to solve:
Apparently the "pattern" property isn't supported anymore. I haven't seen anything like that in the GooCanvas2/GnomeUnstable documentation, so maybe it's something broken on my side?

```
kuhnchris@yumeironosora ~ % cpan -l | grep goo -i  
GooCanvas2::CairoTypes	0.001
GooCanvas2	0.06
kuhnchris@yumeironosora ~ % pacman -Q goocanvas2                                                                       
goocanvas2 2.0.4-9
kuhnchris@yumeironosora ~ % shutter --version
0.96 Rev.1547
```

#### Temporary fix:
comment line 264 and 271.
![image](https://user-images.githubusercontent.com/761911/120925677-f117be00-c6d9-11eb-83fc-aa95e7d9b582.png)

**Comment by DarthGandalf** [comment]
https://developer.gnome.org/goocanvas/unstable/GooCanvasImage.html#GooCanvasImage--pattern

**Comment by DarthGandalf** [comment]
GooCanvas2::CairoTypes by itself is not a problem, it's used only to convert the type of the returned object: `$item->get('pattern')` returns `GooCanvas2::CairoPattern` but `set_filter` is available only on `Cairo::Pattern` even though it's the same underlying object

I don't know why GooCanvas2::CanvasImage doesn't have this property in your case

**Comment by kuhnchris** [comment]
OK, after much fiddling around it seems a perl upgrade may have broken my neck here.
Solution was to force-reinstall GooCanvas2 via cpan (```cpan -fi GooCanvas2 GooCanvas2::CairoTypes```)...
Shall I close this as "user error"?

Thanks!

**Comment by DarthGandalf** [comment]
So it works now without that patch? Great!

The problem with your workaround (not calling set_filter) is a very blurry preview window in the corner

**Comment by Photon89** [comment]
@DarthGandalf But I think, usually packages report such a situation like in 

`Can't locate Gnome2/Canvas.pm in @INC (you may need to install the Gnome2::Canvas module) (@INC contains: /usr/lib/perl5/5.32/site_perl /usr/share/perl5/site_perl /usr/lib/perl5/5.32/vendor_perl /usr/share/perl5/vendor_perl /usr/lib/perl5/5.32/core_perl /usr/share/perl5/core_perl) at (eval 9) line 1. BEGIN failed--compilation aborted at (eval 9) line 1.`

I wonder why this didn't happen here, it makes finding a solution more difficult...

**Comment by kuhnchris** [comment]
> @DarthGandalf But I think, usually packages report such a situation like in
> 
> `Can't locate Gnome2/Canvas.pm in @INC (you may need to install the Gnome2::Canvas module) (@INC contains: /usr/lib/perl5/5.32/site_perl /usr/share/perl5/site_perl /usr/lib/perl5/5.32/vendor_perl /usr/share/perl5/vendor_perl /usr/lib/perl5/5.32/core_perl /usr/share/perl5/core_perl) at (eval 9) line 1. BEGIN failed--compilation aborted at (eval 9) line 1.`
> 
> I wonder why this didn't happen here, it makes finding a solution more difficult...

i *may* assume this may be connected to ArchLinux, Pacman and CPAN in particular - as there *was* a version installed, but it was apparently not working for the upgraded perl version, as in, a 5.32 one was available but the 5.34 one was used, so technically the lib was there but the glue-layer was compiled against the wrong perl version so it caused all kind of funky things. I never used perl that much before so I'm not sure if that is an common issue.

And yes, @DarthGandalf - this issue is solved without the patch - thanks. :-)
It may still make sense to check for availablility of that function tho, as that could make error solving easier later on. But I do agree that this may be a very special corner case and may be ignored. We'll keep it here for documentation purpose, I suppose. ;-)

**Comment by Photon89** [comment]
@kuhnchris I use Arch as well and usually get errors like the one I cited when some perl AUR package hasn't been rebuilt against a new perl version. So, either it is related to the fact that you use CPAN as well (why, actually, all dependencies for Shutter are in the repos or in the AUR, also Shutter itself), or to the particular packages in question.
